conan/tools/meson/toolchain.py

```diff
@@ -442,6 +442,14 @@ def _context(self):
         # These link_args have already the LDFLAGS env value so let's add only the new possible ones
         self.objc_link_args.extend(apple_flags + extra_flags["ldflags"])
         self.objcpp_link_args.extend(apple_flags + extra_flags["ldflags"])
+        # Meson properties
+        sys_root = self._conanfile.conf.get("tools.build:sysroot", check_type=str)
+        if sys_root:
+            self.properties["sys_root"] = sys_root
+            self.c_args.append("--sysroot=" + sys_root)
+            self.cpp_args.append("--sysroot=" + sys_root)
+            self.c_link_args.append("--sysroot=" + sys_root)
+            self.cpp_link_args.append("--sysroot=" + sys_root)
```

Comment:
Instead of adding this here, I'd expect to do it in the very first moment instead. Something like this:

```diff
diff --git a/conan/tools/meson/toolchain.py b/conan/tools/meson/toolchain.py
index aa1d4cb91..6bb0284cb 100644
--- a/conan/tools/meson/toolchain.py
+++ b/conan/tools/meson/toolchain.py
@@ -245,6 +245,10 @@ class MesonToolchain(object):
             default_comp = "cl"
             default_comp_cpp = "cl"

+        # Read configuration for sys_root property (honoring existing conf)
+        self._sys_root = self._conanfile_conf.get("tools.build:sysroot", check_type=str)
+        if self._sys_root:
+            self.properties["sys_root"] = self._sys_root
         # Read configuration for compilers
         compilers_by_conf = self._conanfile_conf.get("tools.build:compiler_executables", default={},
                                                      check_type=dict)
@@ -404,10 +408,11 @@ class MesonToolchain(object):
         linker_scripts = self._conanfile_conf.get("tools.build:linker_scripts", default=[], check_type=list)
         linker_script_flags = ['-T"' + linker_script + '"' for linker_script in linker_scripts]
         defines = [f"-D{d}" for d in self._conanfile.conf.get("tools.build:defines", default=[], check_type=list)]
+        sys_root = [f"--sysroot={self._sys_root}"] if self._sys_root else [""]
         return {
-            "cxxflags": cxxflags,
-            "cflags": cflags,
-            "ldflags": sharedlinkflags + exelinkflags + linker_script_flags,
+            "cxxflags": cxxflags + sys_root,
+            "cflags": cflags + sys_root,
+            "ldflags": sharedlinkflags + exelinkflags + linker_script_flags + sys_root,
             "defines": defines
         }
```

Reply:
@franramirez688 Well observed. I still thinking about the where users inject both `cflags- ... --sysroot` and also uses `tools.build:sysroot` so we would have double sysroot configured. But I think will be a prone error, doesn't?

Reply:
Are there recipes adding this?

Reply:
At least boost adds sysroots in its ld and cflags

Reply:
@franramirez688 No because is about sysroot and cross-building. But still, people can mess by Conan global config.

Reply:
Anyway, in this case, what people expect is something like AutotoolsToolchain is doing:
```python
        sysroot = self._conanfile.conf.get("tools.build:sysroot")
        sysroot = sysroot.replace("\\", "/") if sysroot is not None else None
        self.sysroot_flag = "--sysroot {}".format(sysroot) if sysroot else None
```
So, IMO, it's expected that Meson injects all what's needed.